scripts/airgap.sh

```diff
@@ -0,0 +1,195 @@
+        - $script pull --policy-list policies.txt
+        
+        Push images and policies from created archives to local registry
+        - $script push --policy-list policies.txt --policies policies.tar.gz
+
+        Setup registry with default images and all policies from policies.tar.gz archive
+        - $script setup --policies policies.tar.gz --registry localhost:5000 --kwctl-sources sources.yaml
+      " 1>&2
+}
+
+
+while [[ $# -gt 0 ]]; do
+   case $1 in
+   push|pull|setup)
+      action="$1"
+      shift;;
+   -r|--registry)
+      REGISTRY="$2"
+      shift 2;;
+   -i|--images)
+      IMAGES_TAR="$2"
+      shift 2;;
+   -p|--policies)
+      POLICIES_TAR="$2"
+      shift 2;;
+   --image-list)
+      IMAGES_TXT="$2"
+      shift 2;;
+   --policy-list)
+      POLICIES_TXT="$2"
+      shift 2;;
+   -s|--kwctl-sources)
+      KWCTL_SOURCES="$2"
+      shift 2;;
+   -h|--help)
+      usage
+      exit 0;;
+   *)
+      usage
+      exit 1;;
+   esac
+done
+set -- "${POSITIONAL_ARGS[@]}" # restore positional parameters
+
+trap 'echo "Error on ${BASH_SOURCE}:${LINENO} $(sed -n "${LINENO} s/^\s*//p" ${BASH_SOURCE})"' ERR
+
+# ==================================================================================================
+# Check parameters, requirements and set defaults
+
+# TODO - delme
+: ${KWCTL_SOURCES:=sources.yaml}
+: ${REGISTRY:=localhost:5000}
+
+# Check parameters & Requirement
+[[ "${action:-}" =~ ^(pull|push|setup)$ ]] || { echo "Action not recognized: ${action:--}"; exit 1; }
+[[ "$action" =~ ^(push|setup)$ ]] && : ${REGISTRY:?Required parameter is missing: --registry}
+[ -v REGISTRY ] && curl --output /dev/null --silent --head --fail "$REGISTRY" # check registry is online
+
+[ -v IMAGES_TXT ] && test -f "$IMAGES_TXT"
+[ -v POLICIES_TXT ] && test -f "$POLICIES_TXT"
+[ -v KWCTL_SOURCES ] && test -f "$KWCTL_SOURCES"
+if [ "$action" == "push" ]; then
+   [ -v IMAGES_TAR ] && test -f "$IMAGES_TAR"
+   [ -v POLICIES_TAR ] && test -f "$POLICIES_TAR"
+fi
+which jq kwctl helm docker > /dev/null || { echo "Missing one of requirements"; exit 1; }
+
+# Set defaults
+: ${IMAGES_TAR:=kubewarden-images.tar}
+: ${POLICIES_TAR:=kubewarden-policies.tar.gz}
+
+kwctl_args=${KWCTL_SOURCES:+--sources-path $KWCTL_SOURCES}
+tmpfile=$(mktemp /tmp/kubewarden-airgap.XXXXXX)
+
+# ==================================================================================================
+# Functions
+
+# Print green colored message
+function step() { printf -- "\e[32m${*}\e[0m\n"; }
+
+# Get latest policiy and image lists from artifacthub
+# Use text files from kubewarden-defaults chart
+function pull_latest_lists() {
+   tag=$(helm search hub kubewarden-defaults -o json | jq -er '.[] | select(.repository.url=="https://charts.kubewarden.io").version')
+   assets_urls=$(curl -sL https://api.github.com/repos/kubewarden/helm-charts/releases/tags/kubewarden-defaults-$tag | jq -er '.assets[].browser_download_url')
```

Comment:
I don't like this, the canonical locations of artifacts are either the HTPPS charts.kubewarden.io with unsigned charts, or the OCI registry ghcr.io/kubewarden/* with signed helm charts. GH releases are indeed being consumed by https://charts.kubewarden.io/index.yaml, but I'm not sure we should consume them directly.

Also, given that air gap users are more security-conscious than the average, I expect them to verify the signatures of the helm charts prior to obtaining the images and policies list. Hence why in the current docs there's a "stop gap" between obtaining the helm charts (as the user desires, and verifying them or not), and then using an image list (either from the helm charts or the release).

I would prefer to do a `helm pull` either from one repo or the other, specifed by the user. Then, have a script command that finds the lists and compiles them.

Also note, that the current approach doesn't use the GH release files ala
`kubewarden-crds-1.2.3_kubewarden-defaults-1.2.8_kubewarden-controller-1.2.8_policylist.txt`. I think not using them is indeed more elegant, wonder what the rest think.